components/Profile/SettingsModal.jsx

```diff
@@ -59,31 +61,115 @@ const ModalView = styled(TouchableOpacity)`
   align-items: center;
   shadow-color: #000;
   box-shadow: 0px 2px 4px;
-  shadow-opacity: 0.25px;
+  shadow-opacity: 0.25;
   shadow-radius: 4px;
-  elevation: 5px;
+  elevation: 5;
+`;
+
+const TwinView = styled(View)`
+  display: flex;
+  width: 90%;
+  justify-content: space-between;
+  flex-direction: row;
+  align-items: center;
+  margin-top: 10px;
+`;
+
+const SettingLabel = styled(Text)`
+  font-family: 'Montserrat_500Medium';
+  font-size: 16px;
+  color: ${(props) => (props.disabled ? 'darkgray' : 'black')};
+`;
+
+const SettingTextInput = styled(TextInput)`
+  font-family: 'Montserrat_500Medium';
+  font-size: 16px;
+  color: ${(props) => (props.disabled ? 'darkgray' : 'black')}
+  margin-right: 10px;
+  background-color: #EFEFEF;
+  text-align: center;
+  width: 55px;
+  border-radius: 5px;
 `;
 
 const SettingsModal = (props) => {
   const { triggerSize } = props;
   const [modalVisible, setModalVisible] = useState(false);
   const navigation = useNavigation();
+  const dispatch = useDispatch();
+  const settings = useSelector((state) => state.settings);
+  const [settingState, setSettingState] = useState({
```

Comment:
This is kinda nit-picky, but if you took out the async part in the redux action I think it would simplify this a bit because it would remove the need for the local settings state as well as redux settings. Then maybe when the settings modal is closed it could update to the user profile in the db in the background. Just an idea, this looks like it works fine as is.

Reply:
> I couldn't get it to open on my phone (wifi issue not code) so couldn't test notifs, so I'll wait for someone else to approve this PR, but the code looks good. One thing -- do we not have to worry about asking for permissions when the app first boots up? I think that iOS requires apps to ask for notification permission so that the actual settings in your iphone get edited. I briefly read about it [here](https://docs.expo.io/versions/latest/sdk/permissions/) but not sure if we need them.

Yeah you're right I totally forgot abt that let me add it in rn